src/lib.rs

```diff
@@ -55,18 +62,30 @@ impl UPClientZenoh {
             subscriber_map: Arc::new(Mutex::new(HashMap::new())),
             queryable_map: Arc::new(Mutex::new(HashMap::new())),
             query_map: Arc::new(Mutex::new(HashMap::new())),
+            rpc_callback_map: Arc::new(Mutex::new(HashMap::new())),
             callback_counter: AtomicU64::new(0),
         })
     }
 
+    // The UURI format should be "up/<UAuthority id or ip>/<the rest of UUri>"
     fn to_zenoh_key_string(uri: &UUri) -> Result<String, UStatus> {
         let micro_uuri = MicroUriSerializer::serialize(uri).map_err(|_| {
             UStatus::fail_with_code(
                 UCode::INVALID_ARGUMENT,
                 "Unable to serialize into micro format",
             )
         })?;
-        let micro_zenoh_key: String = micro_uuri
+        let mut micro_zenoh_key = String::from("up/");
```

Comment:
Hi @evshary -- I am not seeing a "special path" here through which a UUri is checked for the format I outlined [in this comment](https://github.com/eclipse-uprotocol/up-client-zenoh-rust/pull/2#discussion_r1497721355) to be able to avoid the MicroUriSerializer failing.

In my mind, this would involve adding the ability to be able to serialize the UAuthority alone. I think Daniel was planning to make the serialization and deserialization more fine grained and implemented per field in the UUri struct, so it'd be good for you to [speak up on that issue](https://github.com/eclipse-uprotocol/up-rust/issues/40#issuecomment-1961013890) with this requirement.

Reply:
Hi @PLeVasseur Apologize for missing the logic of creating "special path" here. I will add it.

About the serialization, [the current implementation](https://github.com/eclipse-uprotocol/up-client-zenoh-rust/pull/2/files/1948b92d038d6f01073d687fa00cf9cabc25b6fc..689bb50c22271f1f5db2aab9d005824aec865d19#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R81) extracts the UAUTHORITY_ADDRESS from the serialized Micro URI directly. It's not hard to do so, since UAuthority is the last four bytes in Mirco URI.
https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uri.adoc#422-remote-micro-uris

Reply:
OK, I got what you're talking about. We need the independent UAuthority serialization if we have the special UUri.
I'll raise the issue and now I'll create a workaround function to do the serialization.